app/submitter/converter.py

```diff
@@ -128,7 +128,7 @@ def build_metadata(metadata: MetadataProxy) -> MetadataType:
 
 def get_optional_payload_properties(
     metadata: MetadataProxy, response_metadata: MutableMapping
-) -> MetadataType:
+) -> dict:
```

Comment:
If this is a `MetadataType` can this instead by fixed by type hinting `payload: MetadataType = {}`? or is it genuinely a different type

Reply:
Cause otherwise the `|` operator will be complained about in mypy as we try to merge two different types..

Reply:
Yeah you're totally right, now I've updated mypy, the only way this could be fixed is doing `payload.update(optional_properties)` instead, and since return types are better as concrete, I think what you've done is better anyway, the `MetadataType` doesn't seem correct, so I'm all for getting rid of it and using dict as you've done 👍 my bad

Reply:
Still, good to hear your opinion and discuss it.